e2e/verify/verify.go

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package verify
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/buger/jsonparser"
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+)
+
+type ctx struct {
+	env            e2e.TestEnv
+	corruptedImage string
+	successImage   string
+}
+
+const successURL = "library://sylabs/tests/verify_success:1.0.1"
+const corruptedURL = "library://sylabs/tests/verify_corrupted:1.0.1"
+
+func getNameJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "Name"}
+}
+
+func getFingerprintJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "Fingerprint"}
+}
+
+func getLocalJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "KeyLocal"}
+}
+
+func getKeyCheckJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "KeyCheck"}
+}
+
+func getDataCheckJSON(keyNum string) []string {
+	return []string{"SignerKeys", keyNum, "Signer", "DataCheck"}
+}
+
+func (c *ctx) singularityVerifyKeyNum(t *testing.T) {
+	keyNumPath := []string{"Signatures"}
+
+	tests := []struct {
+		name         string
+		expectNumOut int64  // Is the expected number of Signatures
+		imageURL     string // Is the URL to the container
+		imagePath    string // Is the path to the container
+		expectExit   int
+	}{
+		{
+			name:         "verify number signers",
+			expectNumOut: 3,
+			imageURL:     corruptedURL,
+			imagePath:    c.corruptedImage,
+			expectExit:   255,
+		},
+		{
+			name:         "verify number signers success container",
+			expectNumOut: 1,
+			imageURL:     successURL,
+			imagePath:    c.successImage,
+			expectExit:   0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			verifyOutput := func(t *testing.T, r *e2e.SingularityCmdResult) {
+				// Get the Signatures and compare it
+				eNum, err := jsonparser.GetInt(r.Stdout, keyNumPath...)
+				if err != nil {
+					t.Fatalf("unable to get expected output from json: %s", err)
+				}
+				if eNum != tt.expectNumOut {
+					t.Fatalf("unexpected failure: got: '%d', expecting: '%d'", eNum, tt.expectNumOut)
+				}
+			}
+
+			// Inspect the container, and get the output
+			e2e.RunSingularity(
+				t,
+				tt.name,
+				e2e.WithPrivileges(false),
+				e2e.WithCommand("verify"),
+				e2e.WithArgs("--json", tt.imagePath),
+				e2e.PreRun(func(t *testing.T) {
+					e2e.PullImage(t, tt.imageURL, tt.imagePath)
+				}),
+				e2e.ExpectExit(tt.expectExit, verifyOutput),
+			)
+		})
+	}
+}
+
+func (c *ctx) singularityVerifySigner(t *testing.T) {
```

Comment:
What do you think about restructuring this that each test case has a slice of keys to check? Looks like then we would have 4 test cases, but we would be checking multiple keys within some of them.